interpreter/core/core.py

```diff
@@ -56,17 +58,37 @@ def __init__(self):
         self.vision = False  # LLM supports vision
 
         # Load config defaults
-        self.extend_config(self.config_file)
+        self.load_config()
 
         # Expose class so people can make new instances
         self.Interpreter = Interpreter
 
-    def extend_config(self, config_path):
-        if self.debug_mode:
-            print(f"Extending configuration from `{config_path}`")
-
-        config = get_config(config_path)
-        self.__dict__.update(config)
+    def load_config(self, profile=None):
+        with open(self.config_file, 'r') as file:
+            config = yaml.safe_load(file)
+
+        # Use default_profile if no specific profile is provided
+        if profile is None:
+            profile = config.get('default_profile', None)
+
+        # Load profile-specific configuration
+        if profile in config:
+            #Load default values first
+            self.update_attributes(config['base'])
+            self.update_attributes(config[profile])
+        else:
+            print(f"Profile '{profile}' not found in the configuration file.")
+            raise
+    
+    def update_attributes(self, config):
```

Comment:
While these methodology of extracting self attributes from the config gives the appearance of DRYer, cleaner code, I think it introduces some issues that outweigh the cleanliness.
If we allow users to directly modify the "self" attributes of the class we potentially expose:

**Maintainability Challenges:** Such direct modifications can complicate code maintenance and updates, risking compatibility issues with existing configurations.

**Validation Concerns:** This method may bypass essential validation checks, possibly leading to inconsistencies or invalid program states.

**Encapsulation Breach:** It goes against the OOP principle of encapsulation, crucial for safeguarding an object's internal state and exposing it only through a defined interface.

**Security Risks:** This could introduce vulnerabilities, particularly with critical functionality attributes, allowing potential unauthorized access or manipulation.

This could be considered an anti-pattern.
The simplest way to handle this would be to use setters and simply have the yaml read something like:
```
base:
  # Settings
  local: False
  auto_run: False
  debug_mode: False
  max_output: 2000
  safe_mode: "off"
  disable_procedures: False
  launch_message: ""
  ```
  
  and then assign them explicity in the set_attributes method like:
  ```
  def set_attributes(self, config):
    if 'local' in config:
        self.set_local(config['local'])
    if 'auto_run' in config:
        self.set_auto_run(config['auto_run'])
    if 'debug_mode' in config:
        self.set_debug_mode(config['debug_mode'])
```

The issue here is that if we have like 80+ attributes, this will get cumbersome.
So the best practice in this case would be to use schema based validation, e.g.

Define a schema model with pydantic:
```
from pydantic import BaseModel

class ConfigSchema(BaseModel):
    local: bool = False
    auto_run: bool = False
    debug_mode: bool = False
    max_output: int = 2000
    safe_mode: str = "off"
    disable_procedures: bool = False
    launch_message: str = ""
    # Define other settings as needed...
    # ...

    # Optionally, include nested configurations or lists if required
```

Update the class to apply the config:
```
import yaml
from pydantic import ValidationError

def load_and_validate_config(file_path: str) -> ConfigSchema:
    with open(file_path, 'r') as file:
        config_data = yaml.safe_load(file)

    try:
        return ConfigSchema(**config_data)
    except ValidationError as e:
        print("Validation Error:", e)
        raise
```

Then apply the config:
```
import yaml
from pydantic import ValidationError

class Interpreter:
    # ... other methods and the __init__ method ...

    def load_config(self, profile=None):
        with open(self.config_file, 'r') as file:
            config_data = yaml.safe_load(file)

        # Use the default profile if no specific profile is provided
        if profile is None:
            profile = config_data.get('default_profile', None)

        try:
            # Validate and load the base configuration
            base_config = ConfigSchema(**config_data.get('base', {}))
            self.apply_config(base_config)

            # Validate and load the profile-specific configuration
            if profile in config_data:
                profile_config = ConfigSchema(**config_data[profile])
                self.apply_config(profile_config)
            else:
                print(f"Profile '{profile}' not found in the configuration file.")
                raise ValueError(f"Profile '{profile}' not found.")
        except ValidationError as e:
            print("Configuration validation error:", e)
            raise

    def apply_config(self, config: ConfigSchema):
        # Apply configuration to the class attributes
        self.local = config.local
        self.auto_run = config.auto_run
        self.debug_mode = config.debug_mode
        self.max_output = config.max_output
        self.safe_mode = config.safe_mode
        self.disable_procedures = config.disable_procedures
        self.launch_message = config.launch_message
        # Apply other settings as needed...
        # ...

```